server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -985,9 +993,9 @@ public static Engine.Index prepareIndex(
         long autoGeneratedIdTimestamp,
         boolean isRetry,
         long ifSeqNo,
-        long ifPrimaryTerm
+        long ifPrimaryTerm,
+        long startTime
```

Comment:
Is this a leftover, I do not see this method used except passing in `System.nanoTime` in tests?

If not, can we rename this to `startTimeNanos`?

Reply:
notice that this method is static, that's why we have to provide the start time. I've renamed it.